src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1629,12 +1629,24 @@ private NonReentrantLock StateLock
         ///
         /// This not intended to be the public API, use Document.WithFrozenPartialSemantics() instead.
         /// </summary>
-        public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(DocumentId documentId, CancellationToken cancellationToken)
+        public SolutionState WithFrozenPartialCompilationIncludingSpecificDocument(DocumentId documentId, ImmutableArray<DocumentId> linkedDocumentIds, CancellationToken cancellationToken)
         {
             try
             {
-                var doc = this.GetRequiredDocumentState(documentId);
-                var tree = doc.GetSyntaxTree(cancellationToken);
+                var originalDoc = this.GetRequiredDocumentState(documentId);
+                var originalTree = originalDoc.GetSyntaxTree(cancellationToken);
+                using var _1 = ArrayBuilder<DocumentState>.GetInstance(out var docBuilder);
```

Comment:
Set the default capacity for this too? Or just have a single list of a tuple since we're enumerating the pair anyways?